build/common/utils.sh

```diff
@@ -103,13 +103,18 @@ function ssl_http_behaviour {
     fi
 }
 
+function domain {
+    eval "python3 -c \"import os;print('.'.join(os.environ['API_DOMAIN'].split('.')[-2:]), end='')\""
```

Comment:
```bash
#!/bin/bash

domain=$1
md=""
p2='^(.*\.)?([^.]+\.[a-z]+\.[a-z][a-z])$'
p3='^(.*\.)?([^.]+\.(com|org|net|int|edu|gov|mil))$'
px='^(.*\.)([a-z]+)$'
# 2 letter country codes
if [[ "$domain" =~ $p2 ]] ; then
    md=${BASH_REMATCH[2]};
# 3 letters legacy domain
elif [[ "$domain" =~ $p3 ]] ; then
    md=${BASH_REMATCH[2]};
# All Other
elif [[ "$domain" =~ $px ]] ; then
    md=${BASH_REMATCH[2]};
fi ;
echo "$md"

```

I tried this script yesterday, but it is not working in shell script. So i used python to get the domain.

Reply:
Right now, I am using:
```
"import os;print('.'.join(os.environ['API_DOMAIN'].split('.')[1:]),
```

Reply:
But shouldn't this thing fail for multi level sub domains like `api.web.openwisp.org` ?

Reply:
Also my code will also fail for `api.openwisp.co.in`.

Reply:
If we are using python and we don't care about adding another dependency, we can use [tldextract](https://pypi.org/project/tldextract/) library which is built for this purpose.

Reply:
`tldextract` is also relying on an database to do this thing.
As of `api.web.openwisp.org`, yes, it will fail, but that's mentioned in the `ENV.md` file that you can't use a multipart subdomain.
As of `api.openwisp.co.in`, no it will not fail, please test it out, someone I know is using such a domain so, to the best of my knowledge it should work! 😄

Reply:
I think we should lean in favour of solutions like https://pypi.org/project/tldextract/ instead of implementing our own unless we can do it better in a simpler way.

In this case it looks like the people working on tldextract have done a great job and the library seems well maintained so definitely we should try it so hopefully we can get around the current limitations.

@codesankalp can you give it a try please?
Also try to have a dedicated script for dealing with this unless is a trivial line of code.